app/controllers/assignments_controller.rb

```diff
@@ -110,7 +141,43 @@ def update
 
     @assignment.assign_attributes(ap)
 
-    if @assignment.save
+
+    # Options are:
+    # None -- leave the (unique, ignored) teamset alone
+    # New -- create solo teams for everyone who's already submitted
+    # Copy -- copy an existing teamset, and create solo teams for anyone who's already submitted
+    # Unique -- nothing to be done; it's already a unique teamset
+    # Use -- nothing to be done; continue to share teamset with another assignment
+    # Clone -- duplicate the teamset and transfer over any submissions
+    
+    @assignment.team_subs = (params["teamset"] != "none")
```

Comment:
Not duplicate, not quite. That's why I documented them both: there are four options upon creation, and (depending on the scenario) up to six distinct possibilities for editing.